addons/website_sale/static/src/js/website_sale_product_configurator.js

```diff
@@ -1,125 +1,133 @@
-            rootProduct: this.rootProduct,
-            isWebsite: true,
-            okButtonText: _t('Proceed to Checkout'),
-            cancelButtonText: _t('Continue Shopping'),
-            title: _t('Add to cart'),
-            context: this._getContext(),
-            forceDialog: this.forceDialog,
-        }).open();
+        const showDialog = await rpc('/website_sale_product_configurator/show_dialog', {
+            product_template_id: this.rootProduct.product_template_id,
+            ptav_ids: this.rootProduct.variant_values,
+            is_product_configured: isOnProductPage,
+        });
+        if (!showDialog) {
+            return this._submitForm();
+        }
+        this.call('dialog', 'add', ProductConfiguratorDialog, {
+            productTemplateId: this.rootProduct.product_template_id,
+            ptavIds: this.rootProduct.variant_values,
+            customAttributeValues: this.rootProduct.product_custom_attribute_values.map(
+                customPtav => ({
+                    ptavId: customPtav.custom_product_template_attribute_value_id,
+                    value: customPtav.custom_value,
+                })
+            ),
+            quantity: this.rootProduct.quantity,
+            soDate: serializeDateTime(DateTime.now()),
+            edit: false,
+            isFrontend: true,
+            options: { isMainProductConfigurable: !isOnProductPage },
+            save: async (mainProduct, optionalProducts, options) => {
+                this._trackProducts([mainProduct, ...optionalProducts]);
 
-        this.optionalProductsModal.on('options_empty', null, this._submitForm.bind(this));
-        this.optionalProductsModal.on('update_quantity', null, this._onOptionsUpdateQuantity.bind(this));
-        this.optionalProductsModal.on('confirm', null, this._onModalSubmit.bind(this, true));
-        this.optionalProductsModal.on('back', null, this._onModalSubmit.bind(this, false));
+                const values = await rpc('/website_sale_product_configurator/update_cart', {
+                    main_product: JSON.stringify(this._serializeProduct(mainProduct)),
+                    optional_products: JSON.stringify(optionalProducts.map(this._serializeProduct)),
+                    ...this._getAdditionalRpcParams(),
+                });
+                if (options.goToCart) {
+                    window.location.pathname = '/shop/cart';
+                } else {
+                    wSaleUtils.updateCartNavBar(values);
+                    wSaleUtils.showCartNotification(this.call.bind(this), values.notification_info);
+                }
+                this._getCombinationInfo($.Event('click', { target: $('#add_to_cart') }));
+            },
+            discard: () => {},
+            ...this._getAdditionalDialogProps(),
+        });
+    },
 
-        return this.optionalProductsModal.opened();
+    // Hook to append additional props in overriding modules.
```

Comment:
nitpicking: Could you switch to proper docstrings? :angel: 

```suggestion
    /**
     * Hook to append additional props in overriding modules.
     *
     * @return ...
     */
```

Reply:
Obviously :) Don't know why I didn't do this initially.